apps/server/src/utilities/network.service.ts

```diff
@@ -0,0 +1,190 @@
+import { execSync } from 'child_process';
+import { Injectable, Logger, OnApplicationShutdown } from '@nestjs/common';
+import * as createMDNSServer from 'multicast-dns';
+import { MulticastDNS, QueryPacket } from 'multicast-dns';
+import { StringAnswer } from 'dns-packet';
+import { RemoteInfo } from 'dgram';
+
+@Injectable()
+export class NetworkService implements OnApplicationShutdown {
+    private readonly logger = new Logger(NetworkService.name);
+
+    private mDNSServer: MulticastDNS | undefined;
+
+    constructor() {
+        this.createMDNSServer();
+    }
+
+    async createMDNSServer() {
+        const localIp = await this.getLocalIp();
+
+        if (!localIp) {
+            this.logger.warn('Couldn\'t determine local IP, mDNS server won\'t be started and simbridge.local will not be available');
+            return;
+        }
+
+        this.logger.log(`Local IP is ${localIp}`);
+
+        this.mDNSServer = createMDNSServer({
+            interface: localIp,
+            multicast: true,
+            reuseAddr: true,
+        });
+
+        this.mDNSServer.on('error', (error) => {
+            this.logger.warn(`mDNS server couldn't be started. Error: ${error.message}`);
+        });
+
+        this.mDNSServer.on('warning', (error) => {
+            this.logger.warn(`mDNS server warning: ${error.message}`);
+        });
+
+        this.mDNSServer.on('ready', () => {
+            this.makeAnnouncement(localIp);
+        });
+
+        this.mDNSServer.on('query', (query, client) => {
+            this.onMDNSQuery(query, client);
+        });
+    }
+
+    makeAnnouncement(localIp: string) {
+        this.logger.log('mDNS server started, simbridge.local is available');
+
+        // First, make two announcements, one second apart (https://www.rfc-editor.org/rfc/rfc6762.html#section-8.3)
+        this.mDNSServer.respond([{
+            name: 'simbridge.local',
+            type: 'A',
+            ttl: 1,
+            flush: true,
+            data: localIp,
+        }]);
+
+        setTimeout(() => {
+            this.mDNSServer.respond([{
+                name: 'simbridge.local',
+                type: 'A',
+                ttl: 1,
+                flush: true,
+                data: localIp,
+            }]);
+        }, 1000);
+    }
+
+    async onMDNSQuery(query: QueryPacket, client: RemoteInfo) {
+        // TODO: Handle AAAA records (https://www.rfc-editor.org/rfc/rfc6762.html#section-6.2) or send NSEC (https://www.rfc-editor.org/rfc/rfc6762.html#section-6.1)
+        if (query.questions.some((q) => q.type === 'A' && q.name === 'simbridge.local')) {
+            this.logger.log(`mDNS query from ${client.address}:${client.port}`);
+
+            // Make sure that the IP is always up-to-date despite DHCP shenanigans
+            const localIp = await this.getLocalIp();
+
+            if (!localIp) {
+                this.logger.warn('Couldn\'t determine the local IP address, no mDNS answer will be sent');
+                return;
```

Comment:
are you sure eslint didn't complain about this?

Reply:
It did not. What did you expect it to complain about?